### ISSUE TITLE
Warning that escapeshellarg removed chars from script arguments

### ISSUE THREAD

**Issue by FrodoTheRingWearer** [Cacti/cacti#461]
Would it be possible to add a warning in the Cacti log about escapeshellarg removing characters from arguments passed to scripts? I just spent an hour tracking down why my shell scripts weren't working properly to find out it is because my community name had an exclamation point in it that was removed before it was saved in the poller_items table.

Thanks!

**Comment by cigamit** [comment]
Can you please provide more details.  Until you do so, it's going to be hard to make any corrections.  Show what was stored in the poller_items table, and what was expected.

**Comment by FrodoTheRingWearer** [comment]
Sorry for being vague. I've got a perl script that polls a server, then sums all the CPU usage for all the CPUs then returns the total CPU's and the total usage (I know aggregates can do this now, I've not switched over). Anyway, in the poller_item table I'd end up with a row in the poller_item table that had the following arg1 column:

> perl E:/Cacti/scripts/tcpu.pl -H "127.0.0.1" -C "CommunityName "

The CommunityName is supposed to be "CommunityName!". All the SNMP queries seem to work fine, but none of the scripts work (Because escapeshellarg removes the exclamation points under windows).

Anyway, I tracked the line down that was causing it. It is line 1209 in ./lib/function.php. It says:

> $full_path = str_replace('<' . $item['data_name'] . '>', escapeshellarg($item['value']), $full_path);

It would just be nice to get a warning instead of things not working correctly. Thank you!
